apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx

```diff
@@ -1,9 +1,15 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+import { getSendOrSwapUrl } from '_app/helpers/getSendOrSwapUrl';
+import { getUSDCurrency, useSuiBalanceInUSDC } from '_app/hooks/useDeepbook';
```

Comment:
Why wouldn't you just use the Sui token price through the API (which we use on the explorer) for this instead of deepbook? Would be way fewer API calls.

Reply:
The Deepbook prices are not going to be exactly like on the API we use on Explorer, since it is based on arb pricings there. We should keep it consistent with whats going on inside deepbook.

Reply:
I think we should be consistent within our products, so if this is true then we probably should update explorer too.

Reply:
Updated to use info from CoinGecko